### PR TITLE
Adding array concat and array function

### PR DESCRIPTION
**Pull request by divyansh-sh**

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java [hunk 1]
```diff
@@ -847,6 +847,13 @@ private void unparseOver(SqlWriter writer, SqlCall call, final int leftPrec,
   private void unparseItem(SqlWriter writer, SqlCall call, final int leftPrec) {
     call.operand(0).unparse(writer, leftPrec, 0);
     final SqlWriter.Frame frame = writer.startList("[", "]");
+
```
Comment (Abhi-210793) [inline core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java:850]: Did you test the mig code with this jar?

Comment (divyansh-sh) [reply core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java:850]: Yes

**Comment by Abhi-210793** [comment]
Review done

**Review by dm-shantanuc — APPROVED** [review]
LGTM

**Review by dm-shantanuc — APPROVED** [review]
LGTM

**Pull request merged**